### PR TITLE
Catch2 standalone exe

### PR DESCRIPTION
**Pull request by gargrahul**

### CODE REVIEW

File: catch/external/Catch2/cmake/Catch2/Catch.cmake [hunk 1]
```diff
@@ -118,15 +118,20 @@ same as the Catch name; see also ``TEST_PREFIX`` and ``TEST_SUFFIX``.
...
+# TARGET_LIST TEST_SET
+function(catch_discover_tests_compile_time_detection TARGET TEST_SET)
   cmake_parse_arguments(
     ""
     ""
     "TEST_PREFIX;TEST_SUFFIX;WORKING_DIRECTORY;TEST_LIST;REPORTER;OUTPUT_DIR;OUTPUT_PREFIX;OUTPUT_SUFFIX"
     "TEST_SPEC;EXTRA_ARGS;PROPERTIES"
     ${ARGN}
   )
+  message("--------------------------------1. Inside discover ${TARGET}")
```
Comment (ajchintz) [inline catch/external/Catch2/cmake/Catch2/Catch.cmake]: Remove these print statements

File: catch/external/Catch2/cmake/Catch2/Catch.cmake [hunk 2]
```diff
@@ -184,66 +207,226 @@ endfunction()
...
+    get_property(crosscompiling_emulator
+      TARGET ${_EXE_NAME}
+      PROPERTY CROSSCOMPILING_EMULATOR
+    )
+    set(_EXE_NAME_LIST ${_EXE_NAME_LIST} ${_EXE_NAME})
+    if(NOT STANDALONE_TESTS EQUAL "1")
+      break()
+    endif()
   endforeach()
+  message("------------------------------------------------AJAY calling discovering tests - ${_EXE_NAME_LIST} ${_NAME}")
```
Comment (ajchintz) [inline catch/external/Catch2/cmake/Catch2/Catch.cmake]: @gargrahul - can you remove this line please

File: catch/external/Catch2/cmake/Catch2/Catch.cmake [hunk 3]
```diff
@@ -184,66 +204,225 @@ endfunction()
...
+    set(_EXE_NAME_LIST ${_EXE_NAME_LIST} ${_EXE_NAME})
+    if(NOT STANDALONE_TESTS EQUAL "1")
+      break()
+    endif()
   endforeach()
+  catch_discover_tests("${_EXE_NAME_LIST}" "${_NAME}" PROPERTIES  SKIP_REGULAR_EXPRESSION "HIP_SKIP_THIS_TEST")
 endfunction()
 
 
+
```
Comment (jatinx) [inline catch/external/Catch2/cmake/Catch2/Catch.cmake]: empty lines?

File: catch/external/Catch2/cmake/Catch2/Catch.cmake [hunk 4]
```diff
@@ -118,8 +118,10 @@ same as the Catch name; see also ``TEST_PREFIX`` and ``TEST_SUFFIX``.
 
 #]=======================================================================]
 
+
 #------------------------------------------------------------------------------
-function(catch_discover_tests TARGET)
+# TARGET_LIST TEST_SET
+function(catch_discover_tests_compile_time_detection TARGET TEST_SET)
```
Comment (jatinx) [inline catch/external/Catch2/cmake/Catch2/Catch.cmake:124]: Can we not change this func name?

This is a standard catch2 macro and will complicate when we decide to update catch2 for HIP tests

File: catch/external/Catch2/cmake/Catch2/Catch.cmake [hunk 5]
```diff
@@ -184,66 +204,225 @@ endfunction()
...
 set(_CATCH_DISCOVER_TESTS_SCRIPT
   ${CMAKE_CURRENT_LIST_DIR}/CatchAddTests.cmake
   CACHE INTERNAL "Catch2 full path to CatchAddTests.cmake helper file"
 )
 
+
 ###############################################################################
 # function to be called by all tests
-function(hip_add_exe_to_target)
+function(hip_add_exe_to_target_compile_time_detection)
```
Comment (jatinx) [inline catch/external/Catch2/cmake/Catch2/Catch.cmake:257]: I dont see this function being used anywhere? am I missing something here?

File: catch/external/Catch2/cmake/Catch2/Catch.cmake (hunk 3 above)
Comment (gargrahul) [reply catch/external/Catch2/cmake/Catch2/Catch.cmake]: Removed these in the latest commit.

File: catch/external/Catch2/cmake/Catch2/Catch.cmake (hunk 5 above)
Comment (gargrahul) [reply catch/external/Catch2/cmake/Catch2/Catch.cmake:257]: Let us discuss more on this  on the internal review as there are some additional changes planned on top of this patch. To unblock the CI, I am approving and merging this PR.

**Review by gandryey — APPROVED** [review]

**Pull request merged**